src/python/py/models/builder.py

```diff
@@ -401,7 +402,7 @@ def save_model(self, out_dir):
 
         # Quantize ONNX model to desired precision
         # TODO: Replace by quantizing the MatMuls as they are created
-        if self.onnx_dtype == "int4" and self.quant_type is None:
+        if self.onnx_dtype == "int4":
             model = self.to_int4(model)
```

Comment:
Is there a check to prevent quantizing twice somewhere? When an AWQ model is an input, is this code not running a quantization on already quantized nodes?

Reply:
In ORT's 4-bits quantizer, the [target op types](https://github.com/microsoft/onnxruntime/blob/7b9db658c3930aec0ea9f8bd550145f87df1f59f/onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py#L1070-L1074) are `MatMul` and `Gather`.

The below check is for which MatMul op(s) to create.

https://github.com/microsoft/onnxruntime-genai/blob/16b11ee9488430da8f0aa78e451d0df33af0e771/src/python/py/models/builder.py#L692-L701

For already-quantized models, `make_matmul_int4` will create a `MatMulNBits` node but `make_matmul_int4_qdq` will create `DequantizeLinear --> Transpose --> MatMul`.

I'll add a check to prevent quantizing the latter `MatMul` again for the already-quantized, use-QDQ case.